### PR TITLE
Constrained Hubbard

### PR DESCRIPTION
**Pull request by mtaillefumier**
This PR implements the Hubbard calculations in SIRIUS using constrained occupancies. The non-colinear case is not covered in this PR (although it will work) because qe does not support it and we should find a way to describe complex numbers using the json format.
The API has one additional call that set which orbital should be constrained.

### CODE REVIEW

**Review by toxa81 — CHANGES_REQUESTED** [review]
There is really one change related to naming of one of the hubbard methods and idntation level of omp pragmas

**Comment by mtaillefumier** [comment]
I do not mind enforcing indentation but please merge the formatting PR first then I can fix all these needy greedy formatting errors.

File: src/context/simulation_parameters.hpp [hunk 1]
```diff
@@ -365,6 +365,11 @@ class Simulation_parameters
         return cfg().parameters().hubbard_correction();
     }
 
+    bool hubbard_constrained_calculation() const
+    {
+      return cfg().hubbard().constrained_hubbard_calculation();
```
[the remark this answers is not in the document]
Comment (mtaillefumier) [reply src/context/simulation_parameters.hpp]: i simplified all new hubbard options.

**Comment by toxa81** [comment]
@mtaillefumier we should do the opposite - merge this first, then apply formatting

**Comment by mtaillefumier** [comment]
I still have two things to fix

- the order of the Ylm between sirius and qe
- q-e-sirius is buggy right now. Some problem with the interface

**Comment by mtaillefumier** [comment]
I think I fixed all remaining issues. pure QE is buggy with original implementation and only converges after 151 iterations while q-e-sirius converges after 31 iterations.

**Comment by toxa81** [comment]
@mtaillefumier ci/cd passes, so it's good from my side. OK to merge?

**Comment by mtaillefumier** [comment]
let's merge.

**Pull request merged**